demo/freedv_700d_rx.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+'''
+  Demo receive program for FreeDV API 700D mode.
+
+  cd ~/codec2/build_linux
+  cat ../raw/ve9qrp_10s.raw | ./demo/freedv_700d_tx | ../demo/freedv_700d_rx.py | aplay -f S16_LE
+
+  Credits: Thanks DJ2LS, xssfox, VK5QI
+'''
+
+import ctypes
+from ctypes import *
+import sys
+import pathlib
+import platform
+
+if platform.system == 'Darwin':
```

Comment:
This should likely be platform.system(), not platform.system:

```
MacBook-Pro:build_osx mooneer$ python3
Python 3.8.7 (default, Jan  1 2021, 16:08:27) 
[Clang 12.0.0 (clang-1200.0.32.28)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import platform
>>> platform.system
<function system at 0x1059d1a60>
>>> platform.system()
'Darwin'
```

Reply:
Also, there should be an additional check for Windows and .dll used instead if true. Not sure how often people will be trying to run the unit tests on Windows, though.

Reply:
> This should likely be platform.system(), not platform.system:

Could you make that change and try it out please?

Reply:
I was able to get farther but it overflows the stack trying to call freedv_get_n_max_speech_samples(). Looks like specifying argument and return types (more info [here](https://docs.python.org/3/library/ctypes.html)) seems to solve that particular crash, but it should probably be done for all of the calls:

```
>>> import ctypes
>>> from ctypes import *
>>> import pathlib
>>> libname = pathlib.Path().absolute() / "src/libcodec2.dylib"
>>> c_lib = ctypes.CDLL(libname)
>>> c_lib.freedv_open.restype = c_void_p
>>> fdv = c_lib.freedv_open(7)
>>> c_lib.freedv_get_n_max_speech_samples.argtypes = [c_void_p]
>>> c_lib.freedv_get_n_max_speech_samples.restype = c_int
>>> c_lib.freedv_get_n_max_speech_samples(fdv)
2560
```

Reply:
Here's the section for [ctypes return-types](https://docs.python.org/3/library/ctypes.html#return-types)
In my opinion it's not necessary for returned integers, which should be the case for freedv_get_n_max_speech_samples() ?
`By default functions are assumed to return the C int type. Other return types can be specified by setting the restype attribute of the function object.`

It would be interesting for me as well, if this is operating system specific, maybe because of different types version - on ubuntu I don't need to declare the variable types 🤔

Reply:
Yeah, I think specifying the argtypes was the more important bit, but it doesn't hurt to explicitly mention return type too (even if it does match the default int type).

Reply:
Thanks @tmiw - curious how it works on Ubuntu.  I did have similar declarations in an earlier version, I'll put them back in.